src/registrar/tests/common.py

```diff
@@ -88,6 +101,226 @@ def send_email(self, *args, **kwargs):
         self.EMAILS_SENT.append({"args": args, "kwargs": kwargs})
 
 
+class AuditedAdminMockData:
+    """Creates simple data mocks for AuditedAdminTest.
+    Can likely be more generalized, but the primary purpose of this class is to simplify
+    mock data creation, especially for lists of items,
+    by making the assumption that for most use cases we don't have to worry about
+    data 'accuracy' ('testy 2' is not an accurate first_name for example), we just care about
+    implementing some kind of patterning, especially with lists of items.
+
+    Two variables are used across multiple functions:
+
+    *item_name* - Used in patterning. Will be appended en masse to multiple string fields,
+    like first_name. For example, item_name 'egg' will return a user object of:
+
+    first_name: 'egg first_name:user',
+    last_name: 'egg last_name:user',
+    username: 'egg username:user'
+
+    where 'user' is the short_hand
+
+    *short_hand* - Used in patterning. Certain fields will have ':{shorthand}' appended to it,
+    as a way to optionally include metadata in the string itself. Can be further expanded on.
+    Came from a bug where different querysets used in testing would effectively be 'anonymized', wherein
+    it would only display a list of types, but not include the variable name.
+    """  # noqa
+
+    # Constants for different domain object types
+    INFORMATION = "information"
+    APPLICATION = "application"
+    INVITATION = "invitation"
+
+    def dummy_user(self, item_name, short_hand):
+        """Creates a dummy user object,
+        but with a shorthand and support for multiple"""
+        user = User.objects.get_or_create(
+            first_name="{} first_name:{}".format(item_name, short_hand),
+            last_name="{} last_name:{}".format(item_name, short_hand),
+            username="{} username:{}".format(item_name, short_hand),
+        )[0]
+        return user
+
+    def dummy_contact(self, item_name, short_hand):
+        """Creates a dummy contact object"""
+        contact = Contact.objects.get_or_create(
+            first_name="{} first_name:{}".format(item_name, short_hand),
+            last_name="{} last_name:{}".format(item_name, short_hand),
+            title="{} title:{}".format(item_name, short_hand),
+            email="{}testy@town.com".format(item_name),
+            phone="(555) 555 5555",
+        )[0]
+        return contact
+
+    def dummy_draft_domain(self, item_name):
+        """Creates a dummy draft domain object"""
+        return DraftDomain.objects.get_or_create(name="city{}.gov".format(item_name))[0]
```

Comment:
If you are modifying the provided name maybe that should be documented in the doc string. For tests this seems like it would hide the name you are making, and I could easily see someone trying to make dummy_domain or whatever with "test.gov" and expect the object to have name "test" only to then see it is "citytest.gov.gov". Seems worth documenting here.

Reply:
That is a good point, I think it may be best if I just remove that entirely and replace it with name=item_name. My original intention was to provide a template to reduce typing but it doesn't really make sense because its a difference of like typing 8 characters vs 1